lib/logstash/codecs/fluent.rb

```diff
@@ -145,4 +148,11 @@ def generate_event(map, fluent_time, tag)
     event
   end
 
+  def map_timestamp(event_hash)
+    event_hash.each do |k, v|
+      event_hash[k] = v.to_iso8601 if v.kind_of?(LogStash::Timestamp)
+      event_hash[k] = map_timestamp(v) if v.kind_of?(Hash)
```

Comment:
Sorry, I should have been more specific - this will still fail if the `LogStash::Timestamp` is in an array as well as a hash - the solution will need to check inside arrays and hashes

Reply:
:disappointed_relieved:
Thanks for catch up! Now `Array` is also supported.
Do we have any other structure which can bring crash possibility?